WordPress/wporg-developer
# Fix compatibility issues with shortcodes in blocks and Gutenberg 18.1+

Fixes #513 

After https://github.com/WordPress/gutenberg/pull/60349, shortcodes passed via block attributes don't work.

Shortcodes in block markup in patterns do work, so this PR removes the custom block usage and puts the shortcodes directly in the pattern markup. The main reason for using these custom blocks was conditional rendering, which has been moved to the template level.

The result is a fair bit of markup duplication but the reduction of custom block usage and hidden complexity is probably positive.

### Testing

1. Ensure your Gutenberg version is 18.1+
2. Load the Plugins handbook. Only the First Published and Last Updated (if different from published) meta data should be displayed
3. Load the Advanced Admin handbook. The dates plus Edit on GitHub and Changelog meta data should be displayed. The GitHub links should include the article title for screen readers only.
4. Load the Block Editor handbook. The meta data should be the same as the Advanced Admin handbook, but without the Changelog link.
5. Test all the links
